### PR TITLE
Add Product scenarios test cases

### PR DESCRIPTION
**Pull request by dulanjalidilmi**
## Purpose
Purpose of this PR is to include the product scenario tests to the product repo.

## Approach
This repo contains sample test case for identified test scenario. 'test.sh' file will be used to extract background information to execute tests and copy reports bucket directory.

## Test environment
This is tested locally in following environment.
    JDK - 1.8
    OS - macOS High Sierra

### CODE REVIEW

File: product-scenarios/1-communicate-in-heterogeneous-message-formats/1.1-convert-soap-messages-to-json/1.1.1-convert-using-payload-factory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/AuthenticateStubUtil.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    /**
+     * Authenticate the given web service stub against the Product user manager. This
+     * will make it possible to use the stub for invoking Product admin services.
+     *
+     * @param stub Axis2 service stub which needs to be authenticated
+     */
+    public static void authenticateStub(String userName, String password, Stub stub) {
+        CarbonUtils.setBasicAccessSecurityHeaders(userName, password, stub._getServiceClient());
+    }
+}
```
Comment (isudana) [inline product-scenarios/1-communicate-in-heterogeneous-message-formats/1.1-convert-soap-messages-to-json/1.1.1-convert-using-payload-factory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/AuthenticateStubUtil.java]: Please add a new line.

File: product-scenarios/1-communicate-in-heterogeneous-message-formats/1.1-convert-soap-messages-to-json/1.1.1-convert-using-payload-factory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioTestBase.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
+package org.wso2.carbon.esb.endpoint.test;
```
Comment (isudana) [inline product-scenarios/1-communicate-in-heterogeneous-message-formats/1.1-convert-soap-messages-to-json/1.1.1-convert-using-payload-factory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioTestBase.java]: Add license header.

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/AuthenticatorClient.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+package org.wso2.carbon.esb.endpoint.test;
+
+import org.apache.axis2.client.Stub;
+import org.apache.axis2.context.ServiceContext;
+import org.apache.axis2.transport.http.HTTPConstants;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.authenticator.stub.AuthenticationAdminStub;
+
+public class AuthenticatorClient {
```
Comment (abeykoon) [inline product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/AuthenticatorClient.java]: Class level comment please.

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/AuthenticatorClient.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+    private static final Log log = LogFactory.getLog(AuthenticatorClient.class);
+    private AuthenticationAdminStub authenticationAdminStub;
+
+    public AuthenticatorClient(String backendUrl) throws Exception {
```
Comment (abeykoon) [inline product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/AuthenticatorClient.java]: Shall we add method comments to all the methods on this class?

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioSampleTest.java [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+import org.wso2.carbon.integration.common.admin.client.CarbonAppUploaderClient;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Calendar;
+import java.util.Properties;
+import javax.activation.DataHandler;
+import javax.activation.FileDataSource;
+
+public class ScenarioSampleTest extends ScenarioTestBase {
```
Comment (abeykoon) [inline product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioSampleTest.java]: Class level comment please.

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioSampleTest.java [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+    private ApplicationAdminClient applicationAdminClient;
+    private final String carFileName = "SOAPToJSONCarbonApplication_1.0.0";
+    int timeout = 20;
+    RequestConfig requestConfig = RequestConfig.custom()
+                                               .setConnectTimeout(timeout * 100)
+                                               .setConnectionRequestTimeout(timeout * 1000)
+                                               .setSocketTimeout(timeout * 1000).build();
+
+    @BeforeClass(alwaysRun = true)
+    public void init() throws Exception {
```
Comment (abeykoon) [inline product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioSampleTest.java]: Method comment stating what is init here?

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioSampleTest.java [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+                                        "  \"LookupCityResult\": {\n" +
+                                        "    \"City\": \"Chicago\",\n" +
+                                        "    \"State\": \"IL\",\n" +
+                                        "    \"Zip\": 60601\n" +
+                                        "  }\n" +
+                                        "}";
+                Assert.assertEquals(response.getStatusLine().getStatusCode(), 200);
+                Assert.assertEquals(IOUtils.toString(response.getEntity().getContent()), responseString);
+            }
+        } catch (IOException e) {
```
Comment (abeykoon) [inline product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioSampleTest.java]: Why do we have an additional catch block here?

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioSampleTest.java [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+                String responseString = "{\n" +
+                                        "  \"Error\": {\n" +
+                                        "    \"message\": \"Error while processing the request\",\n" +
+                                        "    \"code\": \"0\",\n" +
+                                        "    \"description\": \"Error while building message. Error while building Passthrough stream\"\n" +
+                                        "  }\n" +
+                                        "}";
+                Assert.assertEquals(IOUtils.toString(response.getEntity().getContent()), responseString);
+            }
+        } catch (IOException e) {
```
Comment (abeykoon) [inline product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioSampleTest.java]: Why do we have an additional catch block here?

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioSampleTest.java [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+        }
+    }
+
+    @Test(description = "Test HTTP the transformation when a invalid status code is given", enabled = false)
+    public void testMessageTransformationFailure() throws Exception {
+        Assert.assertTrue(false, "This test is intentionally failed!");
+    }
+
+    @AfterClass(description = "Test HTTP the transformation")
+    public void close() throws Exception {
```
Comment (abeykoon) [inline product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioSampleTest.java]: Please complete this method.

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioSampleTest.java [hunk 10]
```diff
@@ -0,0 +1,178 @@
...
+                if (ArrayUtils.contains(applicationList, carFileName)) {
+                    isCarFileDeployed = true;
+                    log.info("car file deployed in " + time + " mills");
+                    return isCarFileDeployed;
+                }
+            }
+            try {
+                Thread.sleep(1000);
+            } catch (InterruptedException e) {
+                //ignore
```
Comment (abeykoon) [inline product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioSampleTest.java]: We should not ignore InterruptedExceptions. 
https://rules.sonarsource.com/java/tag/multi-threading/RSPEC-2142

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioTestBase.java [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Properties;
+
+public class ScenarioTestBase {
```
Comment (abeykoon) [inline product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioTestBase.java]: Class level comment?

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioTestBase.java [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+
+    protected static final int ARTIFACT_DEPLOYMENT_WAIT_TIME_MS = 120000;
+    protected static final String resourceLocation = System.getProperty("test.resources.dir");
+
+    /**
+     * This is a utility method to load the deployment details.
+     * The deployment details are available as key-value pairs in {@link #INFRASTRUCTURE_PROPERTIES},
+     * {@link #DEPLOYMENT_PROPERTIES}, and {@link #JOB_PROPERTIES} under the
+     * {@link #INPUTS_LOCATION}.
+     *
```
Comment (abeykoon) [inline product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioTestBase.java]: Please remove the additional line.

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/AuthenticatorClient.java (hunk 3 above)
Comment (dulanjalidilmi) [reply product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/AuthenticatorClient.java]: Fixed by PR [https://github.com/wso2/product-ei/pull/3466](https://github.com/wso2/product-ei/pull/3466)

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/AuthenticatorClient.java (hunk 4 above)
Comment (dulanjalidilmi) [reply product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/AuthenticatorClient.java]: Fixed by PR [https://github.com/wso2/product-ei/pull/3471](https://github.com/wso2/product-ei/pull/3471)

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioSampleTest.java (hunk 5 above)
Comment (dulanjalidilmi) [reply product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioSampleTest.java]: This file has been removed by PR [https://github.com/wso2/product-ei/pull/2977](https://github.com/wso2/product-ei/pull/2977)

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioSampleTest.java (hunk 6 above)
Comment (dulanjalidilmi) [reply product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioSampleTest.java]: This file has been removed by PR [https://github.com/wso2/product-ei/pull/2977](https://github.com/wso2/product-ei/pull/2977)

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioSampleTest.java (hunk 7 above)
Comment (dulanjalidilmi) [reply product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioSampleTest.java]: This file has been removed by PR [https://github.com/wso2/product-ei/pull/2977](https://github.com/wso2/product-ei/pull/2977)

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioSampleTest.java (hunk 8 above)
Comment (dulanjalidilmi) [reply product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioSampleTest.java]: This file has been removed by PR [https://github.com/wso2/product-ei/pull/2977](https://github.com/wso2/product-ei/pull/2977)

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioSampleTest.java (hunk 9 above)
Comment (dulanjalidilmi) [reply product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioSampleTest.java]: This file has been removed by PR [https://github.com/wso2/product-ei/pull/2977](https://github.com/wso2/product-ei/pull/2977)

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioSampleTest.java (hunk 10 above)
Comment (dulanjalidilmi) [reply product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioSampleTest.java]: This file has been removed by PR [https://github.com/wso2/product-ei/pull/2977](https://github.com/wso2/product-ei/pull/2977)

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioTestBase.java (hunk 11 above)
Comment (dulanjalidilmi) [reply product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioTestBase.java]: This is fixed by [https://github.com/wso2/product-ei/pull/2991](https://github.com/wso2/product-ei/pull/2991)

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioTestBase.java (hunk 12 above)
Comment (dulanjalidilmi) [reply product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioTestBase.java]: This is fixed by PR [https://github.com/wso2/product-ei/pull/3471](https://github.com/wso2/product-ei/pull/3471)

**Pull request merged**